### PR TITLE
Updated TPM attestation APIs to remove BinaryData parameters.

### PR DESCRIPTION
**Pull request by LarryOsterman**
Rework the TPM Attestation APIs as a result of feedback from the TPM attestation team.

The TPM attestation team feels that exposing an attestation request and attestation response structure was a better incarnation of their API surface than exposing the underlying byte array, so reworked the TPM attestation APIs to reflect this change.

# All SDK Contribution checklist:

This checklist is used to make sure that common guidelines for a pull request are followed.
- [ ] **Please open PR in `Draft` mode if it is:**
	- Work in progress or not intended to be merged.
	- Encountering multiple pipeline failures and working on fixes.
- [ ] If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.
- [X] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-net/blob/master/CONTRIBUTING.md).**
- [X] **The pull request does not introduce [breaking changes](https://github.com/dotnet/corefx/blob/master/Documentation/coding-guidelines/breaking-change-rules.md).**

### [General Guidelines and Best Practices](https://github.com/Azure/azure-sdk-for-net/blob/master/CONTRIBUTING.md#general-guidelines)
- [X] Title of the pull request is clear and informative.
- [X] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/master/CONTRIBUTING.md#testing-guidelines)
- [ ] Pull request includes test coverage for the included changes.

### [SDK Generation Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/master/CONTRIBUTING.md#sdk-generation-guidelines)
- [ ] The generate.cmd file for the SDK has been updated with the version of AutoRest, as well as the commitid of your swagger spec or link to the swagger spec, used to generate the code. (Track 2 only)
- [X] The `*.csproj` and `AssemblyInfo.cs` files have been updated with the new version of the SDK. Please double check nuget.org current release version.

### Old outstanding PR cleanup
 Please note:
	If PRs (including draft) has been out for more than 60 days and there are no responses from our query or followups, they will be closed to maintain a concise list for our reviewers.

### CODE REVIEW

File: sdk/attestation/Azure.Security.Attestation/src/Models/TpmAttestationRequest.cs [hunk 1]
```diff
@@ -7,13 +7,13 @@
...
     /// </summary>
     [CodeGenModel("TpmAttestationRequest")]
-    internal partial class TpmAttestationRequest
+    public partial class TpmAttestationRequest
     {
         /// <summary>
-        /// Attestation Response data. See  https://docs.microsoft.com/en-us/azure/attestation/virtualization-based-security-protocol for more details
+        /// Attestation Request data. See  <seealso href="https://docs.microsoft.com/en-us/azure/attestation/virtualization-based-security-protocol"/> for more details
         /// </summary>
         public ReadOnlyMemory<byte> Data { get => Base64Url.Decode(InternalData); set => InternalData = Base64Url.Encode(value.ToArray()); }
```
Comment (jsquire) [inline sdk/attestation/Azure.Security.Attestation/src/Models/TpmAttestationRequest.cs:18]: Should we consider caching the result of the Base64 decode or do we think it unlikely that callers will query the property multiple times?

File: sdk/attestation/Azure.Security.Attestation/src/Models/TpmAttestationResponse.cs [hunk 2]
```diff
@@ -7,13 +7,13 @@
...
     /// </summary>
     [CodeGenModel("TpmAttestationResponse")]
-    internal partial class TpmAttestationResponse
+    public partial class TpmAttestationResponse
     {
         /// <summary>
-        /// Attestation Response data. See  https://docs.microsoft.com/en-us/azure/attestation/virtualization-based-security-protocol for more details
+        /// Attestation Response data. See  <seealso href="https://docs.microsoft.com/en-us/azure/attestation/virtualization-based-security-protocol"/> for more details.
         /// </summary>
         public ReadOnlyMemory<byte> Data { get => Base64Url.Decode(InternalData); }
```
Comment (jsquire) [inline sdk/attestation/Azure.Security.Attestation/src/Models/TpmAttestationResponse.cs:18]: Should we consider caching the result of the Base64 decode or do we think it unlikely that callers will query the property multiple times?

**Review by jsquire — APPROVED** [review]

File: sdk/attestation/Azure.Security.Attestation/src/Models/TpmAttestationRequest.cs (hunk 1 above)
Comment (LarryOsterman) [reply sdk/attestation/Azure.Security.Attestation/src/Models/TpmAttestationRequest.cs]: Given the API pattern expected for this API, the Data property will be queried once and only once on each call, so decoding on every access isn't the worst possible outcome.

**Pull request merged**